web/src/user_groups.ts

```diff
@@ -279,3 +296,55 @@ export function get_realm_user_groups_for_dropdown_list_widget(
 
     return [...system_user_groups, ...user_groups_excluding_system_groups];
 }
+
+export function get_dropdown_options_for_can_manage_group_setting(
+    is_group_creation: boolean,
+    group: UserGroup,
+): UserGroupForDropdownListWidget[] {
+    let dropdown_options = get_realm_user_groups_for_dropdown_list_widget("can_manage_group");
+
+    if (is_group_creation) {
+        for (const dropdown_option of dropdown_options) {
+            dropdown_option.unique_id = dropdown_option.unique_id.toString();
+        }
+        dropdown_options = [
+            ...dropdown_options,
+            {name: "All members of this group", unique_id: "created_group"},
+        ];
+
+        dropdown_options = [
+            ...dropdown_options,
+            {
+                name: "You",
+                unique_id: people.get_user_system_group_name_from_user_id(
+                    people.my_current_user_id(),
+                ),
+            },
+        ];
+    } else {
+        dropdown_options = [
+            ...dropdown_options,
+            {name: "All members of this group", unique_id: group.id},
+        ];
+
+        if (is_user_system_group(group.can_manage_group)) {
+            const group_member = get_user_from_user_system_group(group.can_manage_group);
+            const group_creator_option: UserGroupForDropdownListWidget = {
+                name: "You",
```

Comment:
I am not sure whether this has been discussed or not, but do we want to show the pill (the pill of the acting user itself) instead of "You" here as well. And we show the "You" only if it is set to that option and not after we change it to any other group even if the single user group for that user exists.

I think we can have these discussion on CZO about what the initial UI should be for this feature and what we want to leave it for future.

Reply:
I think these points are quite clear from the discussion [here](https://chat.zulip.org/#narrow/stream/101-design/topic/New.20permissions.20model/near/1625136). Discussion about the options that should be shown in edit user group form is also done there. Let me know if there is anything which should be discussed.

Reply:
So, the discussion mentions that in the "Edit user group" form, the option would be like "{user_icon} user-name" but we currently show "You" as the option. And I don't think we have discussed about showing user pill or just appending a user icon with name. Would be good to get some feedback on design till this get further review from Tim.